packages/schema/src/config/common.ts

```diff
@@ -421,7 +421,7 @@ export default defineUntypedSchema({
         '~~': rootDir,
         '@@': rootDir,
         [basename(assetsDir)]: join(srcDir, assetsDir),
-        [basename(publicDir)]: join(srcDir, publicDir),
+        [basename(publicDir)]: resolve(rootDir, publicDir),
```

Comment:
Nice find! I think:

```suggestion
        [basename(publicDir)]: resolve(srcDir, publicDir),
```

Reply:
Shouldn't the public folder be resolved from the ``rootDir``?

Reply:
Previously `dir.public` was resolved relative to `srcDir`. For backwards compatibility `publicDir` is now fully resolved, so effectively the first argument to `resolve` is ignored. But in the case a user sets it to `some-dir` this will be respected as `<srcDir>/some-dir`.

Reply:
I got it, thanks for explanation.